kotlin/samples/hello-terminal/terminalworkflow/src/main/java/com/squareup/sample/helloterminal/terminalworkflow/TerminalWorkflow.kt

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2019 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.sample.helloterminal.terminalworkflow
+
+import com.squareup.workflow.Workflow
+
+typealias ExitCode = Int
+
+/**
+ * TODO write documentation.
+ */
+typealias TerminalWorkflow = Workflow<TerminalConfig, ExitCode, TerminalRendering>
```

Comment:
Is the use of `typealias` really a good example? We've been using interfaces internally, makes it much easier to see what's going on and so far I've seen no penalties. 

So perhaps your `HelloWorkflow` could instead be `TerminalWorkflowImp : TerminalWorkflow, StatefulWorkflow<...>`?

Reply:
The only reason we're using interfaces internally is because Dagger can't handle kotlin.

Reply:
IIRC that's why we started the conversation that lead to the introduction of the trimmed down Workflow interface. Now that we have it using typealias isn't solving any problems, and I'd argue makess the code harder to follow.

Reply:
I disagree with most of your points, but I don't care enough to continue fighting on this.